lib/pubmed_harvester.rb

```diff
@@ -1,12 +1,11 @@
-
 class PubmedHarvester
   # Searches locally, then ScienceWire, then Pubmed for a publication by PubmedId
   # @param [String] pmid PubmedId
   # @return [Array<Hash>] publication in BibJson format.  Usually for Publication sourcelookup requests
   #   If no publication found, returns an empty Array
   def self.search_all_sources_by_pmid(pmid)
     result = Publication.where(pmid: pmid).to_a # TODO: index manual and batch with pmid?
-    result = Publication.find_by_pmid_in_pub_id_table(pmid) if result.empty?
+    result = [Publication.find_by_pmid_pub_id(pmid)].compact if result.empty?
```

Comment:
This is where things get really weird.

Reply:
Less weird than having `find_by` methods in an AR class that don't return 1 record or nil, imo.

Reply:
Yea, `find_by` does imply it should return a record or nil.  Just a pity the callers now have to do some weird array wrapper stuff or the methods could be renamed to avoid the `find_by`, which the new rubocop is complaining about too.  It's sorta convenient to throw this stuff onto Publication to sorta make it behave like it has a `pmid` or a `doi` field when it doesn't really.  Although I don't love the caller-array munging so much, it's fine.  I suppose thinking about rubocop fixes could lead somewhere like `Publication.get_by_pmid` or `Publication.by_pmid` or  `Publication.for_pmid` and similar equally weird stuff.  Otherwise, it all disappears and callers have to do the joins themselves.  Dunno if a `through` could be mangled either because the PublicationIdentifiers don't have any `pmid` or `doi` fields either; maybe a through could have a block on it, dunno.